lightningd/coin_mvts.c

```diff
@@ -1,24 +1,61 @@
 #include <lightningd/coin_mvts.h>
 #include <lightningd/notification.h>
 
+static s64 update_count(struct lightningd *ld)
+{
+	s64 count;
+	bool db_in_tx;
+	db_in_tx = db_in_transaction(ld->wallet->db);
+	if (!db_in_tx)
+		db_begin_transaction(ld->wallet->db);
+	/* This could be written more concisely as
+	 *     count = ++ld->coin_moves_count;
+	 * however I believe that's contra-code conventions */
+	ld->coin_moves_count++;
+	count = ld->coin_moves_count;
+	db_set_intvar(ld->wallet->db, "coin_moves_count",
+		      count);
```

Comment:
No, I have no problem with `++` prefix when necessary (it's just that C++ programmers have been trained to use it by default).  I'd just do this all as a one-liner...